onnx/backend/test/runner/__init__.py

```diff
@@ -295,21 +295,32 @@ def run(test_self, device):  # type: (Any, Text) -> None
                 inputs_num = len(glob.glob(os.path.join(test_data_dir, 'input_*.pb')))
                 for i in range(inputs_num):
                     input_file = os.path.join(test_data_dir, 'input_{}.pb'.format(i))
-                    tensor = onnx.TensorProto()
-                    with open(input_file, 'rb') as f:
-                        tensor.ParseFromString(f.read())
-                    inputs.append(numpy_helper.to_array(tensor))
+                    self._load_proto(input_file, inputs)
```

Comment:
I think that the best approach is to get the types of inputs from the model (read above) and use the types to determine how to load the data. See, for example, here: https://github.com/onnx/onnx/blob/b353e8ba02539ace81dcb762dc78ff4df27cd9b8/onnx/backend/test/cpp/driver/test_driver.cc#L208

Reply:
Makes sense. However, it's hard to match the model.input/output to protobuf files (before knowing the right type and loading)... So I assume the order of model.input/output is same as the order of input/output proto files.

Reply:
I don't understand your concern. The code looks fine to me, since for the i-th input/output we generate the corresponding file name, and get the type from graph.input and graph.output.